internal/common/common.go

```diff
@@ -73,6 +73,92 @@ type ReferenceValues struct {
 	BuilderImageSHA256Digests []string `toml:"builder_image_sha256_digests"`
 }
 
+// ProvenanceIR is an internal intermediate representation of data from provenances.
+// We want to map different provenances of different build types to ProvenanceIR, so
+// all fields except for `binarySHA256Digest` are optional.
+type ProvenanceIR struct {
+	binarySHA256Digest       string
+	buildCmd                 []string
+	builderImageSHA256Digest string
+}
+
+// NewProvenanceIR creates a new proveance with given optional fields.
+// Every provenancy needs to have binary sha256 digest, so this is not optional.
+func NewProvenanceIR(binarySHA256Digest string, options ...func(p *ProvenanceIR)) *ProvenanceIR {
+	provenance := &ProvenanceIR{binarySHA256Digest: binarySHA256Digest}
+	for _, addOption := range options {
+		addOption(provenance)
+	}
+	return provenance
+}
+
+// WithBuildCmd adds a build cmd when creating a new ProvenanceIR.
+func WithBuildCmd(buildCmd []string) func(p *ProvenanceIR) {
+	return func(p *ProvenanceIR) {
+		p.buildCmd = buildCmd
+	}
+}
+
+// WithBuilderImageSHA256Digest adds a builder image sha256 digest when creating a new ProvenanceIR.
+func WithBuilderImageSHA256Digest(builderImageSHA256Digest string) func(p *ProvenanceIR) {
+	return func(p *ProvenanceIR) {
+		p.builderImageSHA256Digest = builderImageSHA256Digest
+	}
+}
+
+// GetBinarySHA256Digest gets the binary sha256 digest. Throws an error if the binary sha256 digest is empty.
+func (p *ProvenanceIR) GetBinarySHA256Digest() (string, error) {
```

Comment:
Apparently the Golang convention is to drop `Get`. It is not followed everywhere in the code yet. Just thought I'd mention it if you are interested. But actually, it might be better to fix this in the entire codebase in a separate PR.

Reply:
Sounds good!